### PR TITLE
Replace Travis with Github Workflow

### PR DESCRIPTION
**Pull request by vtomole**

### CODE REVIEW

File: dev_tools/run_doctest.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            try_print('.', end='', flush=True)
+
+    try_print()
+
+    return doctest.TestResults(failed=failed,
+                               attempted=attempted), error_messages
+
+
+def import_file(file_path: str) -> ModuleType:
+    """Finds and runs a python file as if were imported with an `import`
```
Comment (viathor) [inline dev_tools/run_doctest.py:187]: nit: "...as if **it** were imported..."

File: dev_tools/run_doctest.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    statement.
+
+    Args:
+        file_path: The file to import.
+
+    Returns: The imported module.
```
Comment (viathor) [inline dev_tools/run_doctest.py:193]: nit: newline after colon

File: dev_tools/conf/pip-install-minimal-for-pytest-changed-files.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Install usual requirements.
+pip install -r requirements.txt
+
+# Install pytest related dev requirements.
+cat dev_tools/conf/pip-list-dev-tools.txt | grep pytest | xargs pip install
+
+# Install contrib requirements only if needed.
+changed=$(git diff --name-only origin/master | grep "cirq/contrib" || true)
+[ "${changed}" = "" ] || pip install -r cirq/contrib/contrib-requirements.txt
```
Comment (viathor) [inline dev_tools/conf/pip-install-minimal-for-pytest-changed-files.sh:31]: I think this can go since OpenFermion-Cirq doesn't have contrib directory.

**Comment by kevinsung** [comment]
Apparently the Windows pytest check only takes one second and prints no output.
pytest

File: check/doctest [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+#
+# The -q argument suppresses all output except the final result line and any
+# error messages.
+################################################################################
+
+# Get the working directory to the repo root.
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$(git rev-parse --show-toplevel)"
+
+PYTHONPATH="$(pwd)" python dev_tools/run_doctest.py $@
```
Comment (viathor) [inline check/doctest:20]: nit: it's safer with quotes: `"$@"` (note that bash is smart when quotes are used with `$@` - the result is equivalent to quoting each element in `$@`, not to coalescing all array elements into one quotes string - which wouldn't do what we need)

**Comment by vtomole** [comment]
@kevinsung Some code is breaking the windows tests. Let's merge this without the Windows check https://github.com/quantumlib/OpenFermion-Cirq/issues/397

File: dev_tools/conf/pip-install-minimal-for-pytest-changed-files.sh (hunk 3 above)
Comment (vtomole) [reply dev_tools/conf/pip-install-minimal-for-pytest-changed-files.sh:31]: Done.

File: check/doctest (hunk 4 above)
Comment (vtomole) [reply check/doctest:20]: Done

File: dev_tools/run_doctest.py (hunk 2 above)
Comment (vtomole) [reply dev_tools/run_doctest.py:193]: Done

File: dev_tools/run_doctest.py (hunk 1 above)
Comment (vtomole) [reply dev_tools/run_doctest.py:187]: Done.

**Comment by kevinsung** [comment]
Ok, once all the checks pass, I'll disable the Travis check and merge this.

**Comment by vtomole** [comment]
@kevinsung Checks pass now.

File: dev_tools/run_doctest.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    return mod
+
+
+def main():
+    quiet = len(sys.argv) >= 2 and sys.argv[1] == '-q'
+
+    file_names = glob.glob('cirq/**/*.py', recursive=True)
+    # Remove the engine client code.
+    file_names = [
+        f for f in file_names if not f.startswith('cirq/google/engine/client/')
```
Comment (viathor) [inline dev_tools/run_doctest.py:213]: This can go.

File: dev_tools/run_doctest.py [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    mod = importlib.util.module_from_spec(spec)
+    # Run the code in the module (but not with __name__ == '__main__')
+    spec.loader.exec_module(mod)  # type: ignore
+    return mod
+
+
+def main():
+    quiet = len(sys.argv) >= 2 and sys.argv[1] == '-q'
+
+    file_names = glob.glob('cirq/**/*.py', recursive=True)
```
Comment (viathor) [inline dev_tools/run_doctest.py:210]: This should be `openfermioncirq/**/*.py`.

Comment (vtomole) [reply dev_tools/run_doctest.py:210]: Done

File: dev_tools/run_doctest.py (hunk 5 above)
Comment (vtomole) [reply dev_tools/run_doctest.py:213]: Done.

**Comment by vtomole** [comment]
@viathor Thanks for the reminder. I don't think i've missed anything this time.

**Pull request merged**